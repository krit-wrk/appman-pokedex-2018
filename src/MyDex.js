import React from 'react'
function MyDex(props) {
        return (
                <div>
                        MyDex
                </div>
        )
}