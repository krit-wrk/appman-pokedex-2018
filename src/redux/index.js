export { default as store } from "./store";
export { default as myDexAction } from "./myDex/myDexAction";
