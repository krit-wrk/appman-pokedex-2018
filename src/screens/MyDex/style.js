import { StyleSheet } from "aphrodite";
import { COLORS } from "../../constants";
export default StyleSheet.create({
	container: {
		display: "flex",
		flex: 1,
		flexDirection: "column",
	},
	titleContainer: {
		height: "100px",
		marginTop: "16px",
		fontSize: "50px",
		textAlign: "center",
	},
	dexContainer: {
		display: "flex",
		flex: 1,
		flexWrap: "wrap",
		flexDirection: "row",
		overflowX: "scroll",
		alignContent: "flex-start",
	},
	footer: {
		display: "flex",
		flexDirection: "row",
		// flex: 1,
		height: "60px",
		width: "100%",
		bottom: 0,
		backgroundColor: COLORS.BottomBarBackground,
		justifyContent: "center",
	},
	add: {
		marginTop: "-40px",
		height: "100px",
		width: "100px",
		borderRadius: "50px",
		backgroundColor: COLORS.BottomBarBackground,
		boxShadow: `inset 0 10px 5px -0px ${COLORS.BottomBarBoxShadow}`,
		justifyContent: "center",
		textAlign: "center",
		fontSize: "60px",
		color: COLORS.BottomBarTextColor,
		zIndex: 100,
	},
	card: {
		width: "45%",
		margin: "16px",
	},

	removeButton: {
		textAlign: "center",
		fontSize: "25px",
		color: COLORS.AddButton,
	},
});
