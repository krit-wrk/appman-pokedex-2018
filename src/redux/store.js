import { createStore, combineReducers } from "redux";

import myDexReducer from "./myDex/myDexReducer";

const rootReducer = combineReducers({
	myDex: myDexReducer,
});

const store = createStore(rootReducer);

export default store;
