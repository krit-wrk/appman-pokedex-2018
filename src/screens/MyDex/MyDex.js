import React from 'react'
import { Card } from '../../components'
import style from './style'
import { css } from 'aphrodite'
import { Modal } from "react-responsive-modal";
const MyDex = () => {
    const [isOpen, setIsOpen] = React.useState(false);


    return (
        <div className={css(style.container)} >

            <div onClick={() => setIsOpen(true)} className={css(style.titleContainer)} >
                My Pokedex
      </div>
            <div className={css(style.dexContainer)}>
                <div className={css(style.card)}>
                    <Card
                        name="Deoxys ex"
                        happy={3}
                        hp={30}
                        image={"https://images.pokemontcg.io/ex8/98.png"}
                        str={40}
                        weak={90} />
                </div>
            </div>

            <Modal
                styles={{
                    modal: { margin: 0, padding: 0 }
                }}
                open={isOpen}
                onClose={setIsOpen}
                showCloseIcon={false}

            >
                <div className={css(style.card)}>
                    <Card
                        name="Deoxys ex"
                        happy={3}
                        hp={30}
                        image={"https://images.pokemontcg.io/ex8/98.png"}
                        str={40}
                        weak={90} />
                </div>
            </Modal>

        </div>
    )

}

export { MyDex }
