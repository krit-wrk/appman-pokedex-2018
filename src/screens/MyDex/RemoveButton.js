import React from "react";
import { css } from "aphrodite";
import style from "./style";

export const RemoveButton = ({ onClick }) => (
	<div className={css(style.removeButton)} onClick={onClick}>
		X
	</div>
);
