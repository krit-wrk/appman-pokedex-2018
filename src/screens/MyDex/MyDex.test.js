import React from "react";
import { Provider } from "react-redux";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import axios from "axios";
import { MyDex } from "./MyDex";
import { StyleSheetTestUtils } from "aphrodite";
import { store, myDexAction } from "../../redux";

const getMockData = () => ({
	id: "ex8-98",
	name: "Deoxys ex",
	nationalPokedexNumber: 386,
	imageUrl: "https://images.pokemontcg.io/ex8/98.png",
	imageUrlHiRes: "https://images.pokemontcg.io/ex8/98_hires.png",
	supertype: "Pokémon",
	subtype: "EX",
	ability: {
		name: "Form Change",
		text:
			"Once during your turn (before your attack), you may search your deck for another Deoxys ex and switch it with Deoxys ex. (Any cards attached to Deoxys ex, damage counters, Special Conditions, and effects on it are now on the new Pokémon.) If you do, put Deoxys ex on top of your deck. Shuffle your deck afterward. You can't use more than 1 Form Change Poké-Power each turn.",
		type: "Poké-Power",
	},
	hp: "110",
	retreatCost: ["Colorless", "Colorless"],
	convertedRetreatCost: 2,
	number: "98",
	artist: "Mitsuhiro Arita",
	rarity: "Rare Holo EX",
	series: "EX",
	set: "Deoxys",
	setCode: "ex8",
	text: [
		"When Pokémon-ex has been Knocked Out, your opponent takes 2 Prize cards.",
	],
	attacks: [
		{
			cost: ["Psychic", "Colorless", "Colorless"],
			name: "Psychic Burst",
			text:
				"You may discard 2 Energy attached to Deoxys ex. If you do, this attack does 50 damage plus 20 more damage for each Energy attached to the Defending Pokémon.",
			damage: "50+",
			convertedEnergyCost: 3,
		},
	],
	weaknesses: [
		{
			type: "Psychic",
			value: "×2",
		},
	],
	type: "Psychic",
});

describe("MyDex", () => {
	beforeEach(() => {
		StyleSheetTestUtils.suppressStyleInjection();
		store.dispatch(myDexAction.reset());
	});
	afterEach(() => {
		StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
	});
	it("renders without crashing", async () => {
		const { container } = render(
			<Provider store={store}>
				<MyDex />
			</Provider>,
		);
		expect(container).toMatchSnapshot();
	});
	it("remove ok", async () => {
		const { container, getByText, queryByText } = render(
			<Provider store={store}>
				<MyDex />
			</Provider>,
		);
		expect(container).toMatchSnapshot();
		store.dispatch(myDexAction.add(getMockData()));
		await waitFor(async () =>
			expect(getByText("DEOXYS EX")).toBeInTheDocument(),
		);
		fireEvent.mouseEnter(getByText("DEOXYS EX"));
		const x = getByText("X");
		expect(x).toBeInTheDocument();
		expect(container).toMatchSnapshot();
		fireEvent.click(x);
		expect(queryByText("DEOXYS EX")).not.toBeInTheDocument();
		expect(container).toMatchSnapshot();
		expect(store.getState().myDex.myDex.length).toEqual(0);
		expect(store.getState().myDex.myDexDisplay.length).toEqual(0);
	});
});
