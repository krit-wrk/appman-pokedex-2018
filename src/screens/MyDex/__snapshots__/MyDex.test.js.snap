// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDex remove ok 1`] = `
<div>
  <div
    class="container_107ffsm"
  >
    <div
      class="titleContainer_5agzlf"
    >
      My Pokedex
    </div>
    <div
      class="dexContainer_16xdx4s"
    />
    <div
      class="footer_ciqk7e"
    >
      <div
        class="add_x41nhl"
      >
        +
      </div>
    </div>
  </div>
</div>
`;

exports[`MyDex remove ok 2`] = `
<div>
  <div
    class="container_107ffsm"
  >
    <div
      class="titleContainer_5agzlf"
    >
      My Pokedex
    </div>
    <div
      class="dexContainer_16xdx4s"
    >
      <div
        class="card_1fxyoak"
      >
        <div
          class="container_b4ahsr"
        >
          <div
            class="imageContainer_1g6vvub"
          >
            <img
              alt="Deoxys ex"
              class="image_5hddva"
              src="https://images.pokemontcg.io/ex8/98.png"
            />
          </div>
          <div
            class="desc_19c633u"
          >
            <div
              class="titleContainer_1ryugy2"
            >
              <div
                class="name_4p276u"
              >
                DEOXYS EX
              </div>
              <div>
                <div
                  class="removeButton_5gyw2"
                >
                  X
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                HP
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                STR
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_1hfu0v"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                WEAK
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="cuteContainer_zz3svu"
            >
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer_ciqk7e"
    >
      <div
        class="add_x41nhl"
      >
        +
      </div>
    </div>
  </div>
</div>
`;

exports[`MyDex remove ok 3`] = `
<div>
  <div
    class="container_107ffsm"
  >
    <div
      class="titleContainer_5agzlf"
    >
      My Pokedex
    </div>
    <div
      class="dexContainer_16xdx4s"
    />
    <div
      class="footer_ciqk7e"
    >
      <div
        class="add_x41nhl"
      >
        +
      </div>
    </div>
  </div>
</div>
`;

exports[`MyDex renders without crashing 1`] = `
<div>
  <div
    class="container_107ffsm"
  >
    <div
      class="titleContainer_5agzlf"
    >
      My Pokedex
    </div>
    <div
      class="dexContainer_16xdx4s"
    />
    <div
      class="footer_ciqk7e"
    >
      <div
        class="add_x41nhl"
      >
        +
      </div>
    </div>
  </div>
</div>
`;
